project/backend/src/main/java/com/google/sps/servlets/SwipeMatchServlet.java

```diff
@@ -0,0 +1,124 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.LinkedBlockingQueue;
+import javax.servlet.AsyncContext;
+import javax.servlet.AsyncEvent;
+import javax.servlet.AsyncListener;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(urlPatterns = "/api/swipematch", asyncSupported = true)
+public class SwipeMatchServlet extends HttpServlet {
+  private boolean running;
+
+  private Map<String, AsyncContext> asyncContexts = new ConcurrentHashMap<>();
+
+  private BlockingQueue<String> queue = new LinkedBlockingQueue<>();
+
+  private List<String> store = new CopyOnWriteArrayList<>();
+
+  private Thread notifier = new Thread(() -> {
+    while (running) {
+      try {
+        String user = queue.take();
+
+        store.add(user);
+
+        for (AsyncContext asyncContext : asyncContexts.values()) {
+          try {
+            sendUser(asyncContext.getResponse().getWriter(), user);
+          } catch (Exception e) {
+            asyncContexts.values().remove(asyncContext);
+          }
+        }
+      } catch (InterruptedException e) {
+        // idk yet
```

Comment:
What is an example of when this block would execute (to help us figure out how to handle the exception)?